backend/test/services.terminals.spec.js

```diff
@@ -0,0 +1,577 @@
+              return body
+            }
+          }
+        },
+        extensions: {
+          ingresses: {
+            mergePatch (namespace, name, body) {
+              expect(namespace).to.equal(`shoot--garden--${seedName}`)
+              expect(name).to.equal('dashboard-terminal-kube-apiserver')
+              return body
+            }
+          }
+        }
+      }
+
+      const seedClient = {
+        cluster: {
+          server: new URL('https://apiServerHostname:6443')
+        },
+        core: {
+          services: {
+            mergePatch (namespace, name, body) {
+              expect(namespace).to.equal('shoot--foo--baz')
+              expect(name).to.equal('dashboard-terminal-kube-apiserver')
+              return body
+            }
+          }
+        },
+        extensions: {
+          ingresses: {
+            mergePatch (namespace, name, body) {
+              expect(namespace).to.equal('shoot--foo--baz')
+              expect(name).to.equal('dashboard-terminal-kube-apiserver')
+              return body
+            }
+          }
+        }
+      }
+
+      const shootList = [{
+        kind: 'Shoot',
+        metadata: {
+          namespace: 'garden-foo',
+          name: 'bar'
+        },
+        status: {
+          seed: seedName,
+          technicalID: `shoot--foo--bar`,
+          lastOperation: {
+            progress: 5
+          }
+        }
+      }, {
+        kind: 'Shoot',
+        metadata: {
+          namespace: 'garden-foo',
+          name: 'baz'
+        },
+        status: {
+          seed: seedName,
+          technicalID: `shoot--foo--baz`,
+          lastOperation: {
+            progress: 50
+          }
+        }
+      }]
+
+      beforeEach(function () {
+        sandbox.stub(dashboardClient, 'core').get(() => {
+          return client['core']
+        })
+        sandbox.stub(dashboardClient, 'core.gardener.cloud').get(() => {
+          return {
+            seeds: {
+              get (name) {
+                return getSeed(name)
+              }
+            },
+            shoots: {
+              get (namespace, name) {
+                return _.find(shootList, ({ metadata }) => metadata.name === name && metadata.namespace === namespace)
+              }
+            }
+          }
+        })
+        sandbox.stub(dashboardClient, 'getShoot').callsFake(({ namespace, name }) => {
+          return client.getShoot({ namespace, name })
+        })
+        sandbox.stub(dashboardClient, 'createKubeconfigClient').callsFake(({ namespace, name }) => {
+          if (name === firstSecretName) {
+            return hostClient
+          }
+          if (name === `seedsecret-${soilName}`) {
+            return soilClient
+          }
+          if (name === `seedsecret-${seedName}`) {
+            return seedClient
+          }
+        })
+      })
+
+      it('should not bootstrap anything', async function () {
+        const bootstrap = {
+          disabled: true
```

Comment:
in addition, bootstrapResource seed and shoot -> should not be added to queue

Reply:
✔️